types/validation.go

```diff
@@ -49,6 +56,11 @@ func VerifyCommit(chainID string, vals *ValidatorSet, blockID BlockID,
 			votingPowerNeeded, ignore, count, true, true)
 	}
 
+	// attempt to verify aggregated commit
+	if isAggregatedCommit(vals) {
+		return verifyAggregatedCommit(chainID, vals, commit,
+			votingPowerNeeded, ignore, count, true)
+	}
```

Comment:
This should probably go between lines 52 and 53 above, as the batch verify only makes sense with good-old non-aggregated signatures.